core/transport/websocket_transport.go

```diff
@@ -138,8 +142,12 @@ func (ws *wsServerTransport) putTransport(tp *Transport) bool {
 	}
 }
 
+func NewWebsocketServerTransportWithMux(f ListenerFactory, path string, upgrader *websocket.Upgrader, mux *http.ServeMux) ServerTransport {
+	return NewWebsocketServerTransport(f, path, upgrader, mux)
+}
+
 // NewWebsocketServerTransport creates a new server-side transport.
-func NewWebsocketServerTransport(f ListenerFactory, path string, upgrader *websocket.Upgrader) ServerTransport {
+func NewWebsocketServerTransport(f ListenerFactory, path string, upgrader *websocket.Upgrader, mux *http.ServeMux) ServerTransport {
```

Comment:
It's better to keep the original  `NewWebsocketServerTransport` method signature, I think.


For example, extract a private method:
```go

func innerNewWebsocketServerTransport(f ListenerFactory, path string, upgrader *websocket.Upgrader, mux *http.ServeMux) ServerTransport {
      // original codes of NewWebsocketServerTransport
}

func NewWebsocketServerTransportWithMux(f ListenerFactory, path string, upgrader *websocket.Upgrader, mux *http.ServeMux) ServerTransport {
    return innerNewWebsocketServerTransport(f,path,upgrader,mux)
}

func NewWebsocketServerTransport(f ListenerFactory, path string, upgrader *websocket.Upgrader) ServerTransport {
    return innerNewWebsocketServerTransport(f,path,upgrader,nil)
}

```